app/assets/javascripts/discourse/app/components/modal/associate-account-confirm.js

```diff
@@ -0,0 +1,32 @@
+import Component from "@glimmer/component";
+import { action } from "@ember/object";
+import { inject as service } from "@ember/service";
+import { ajax } from "discourse/lib/ajax";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+
+export default class AssociateAccountConfirm extends Component {
+  @service router;
+  @service currentUser;
+
+  @action
+  async finishConnect() {
+    try {
+      const result = await ajax({
+        url: `/associate/${encodeURIComponent(this.args.model.token)}`,
+        type: "POST",
+      });
+
+      if (result.success) {
+        this.router.transitionTo(
+          "preferences.account",
+          this.currentUser.findDetails()
+        );
+        this.args.closeModal();
+      } else {
+        this.flash = result.error;
```

Comment:
I think we'll need an `@tracked flash` on the class, and we need to pass the `@flash` argument to `<DModal` 👀